source/about/product.rst

```diff
@@ -1,169 +1,56 @@
-Product Overview
-=================
+===================
+Mattermost Overview
+===================
 
```

Comment:
Can be a separate PR, but it would be great to provide at least one screenshot on this page of what Mattermost looks like. Either an "overview" screenshot near the top of the page, or individual screenshots for Channels, Playbooks and Boards in their respective sections.

Perhaps there is something polished Marketing plans to use on their website that can be re-used here?

(Just something I felt was missing on this page)